src/Illuminate/Cache/Repository.php

```diff
@@ -496,7 +496,7 @@ public function flexible($key, $ttl, $callback, $lock = null)
             ], $ttl[1]));
         }
 
-        if (($created + $this->getSeconds($ttl[0])) > Carbon::now()->getTimestamp()) {
+        if (Carbon::now()->getTimestamp() < ($created + $this->getSeconds($ttl[0]))) {
```

Comment:
I've added a test to make sure the boundary is correct...but my brain had to change the order of these so that I could actually say it was correct by just reading it. Feel free to revert.